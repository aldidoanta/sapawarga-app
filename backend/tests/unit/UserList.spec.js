import { shallowMount } from '@vue/test-utils'
import UserList from '@/views/user/list'

describe('UserList', () => {
  const build = () => {
    // arrange
    const wrapper = shallowMount(UserList)
    return {
      wrapper
    }
  }

  it('renders the component', () => {
    const { wrapper } = build()
    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })
})
