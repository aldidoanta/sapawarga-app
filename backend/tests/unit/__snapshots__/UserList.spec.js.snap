// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders the component 1`] = `
<div class="app-container">
  <panel-group-stub totalalluser="1000" totaluserprovince="20" totaluserkabkota="0" totaluserkec="627" totaluserkel="22" totaluserrw="111" duration="2600" total-user-kota="27"></panel-group-stub>
  <el-row-stub tag="div" gutter="20" justify="start" align="top">
    <el-col-stub span="24" tag="div">
      <el-row-stub tag="div" justify="start" align="top" style="margin: 10px 0px;">
        <el-col-stub span="12" tag="div">
          <router-link-stub to="[object Object]" tag="a" event="click">
            <el-button-stub type="primary" size="small" icon="el-icon-plus" nativetype="button">
              Tambah Pengguna Baru
            </el-button-stub>
          </router-link-stub>
        </el-col-stub>
      </el-row-stub>
      <listfilter-stub listquery="[object Object]"></listfilter-stub>
      <el-table-stub fit="true" stripe="true" border="true" showheader="true" highlightcurrentrow="true" selectonindeterminate="true" indent="16" style="width: 100%;">
        <el-table-column-stub type="index" width="50" resizable="true" align="center" filtermultiple="true" sortorders="ascending,descending,"></el-table-column-stub>
        <el-table-column-stub type="default" label="Nama Pengguna" prop="name" sortable="custom" resizable="true" filtermultiple="true" sortorders="ascending,descending,"></el-table-column-stub>
        <el-table-column-stub type="default" label="Kedudukan" resizable="true" filtermultiple="true" sortorders="ascending,descending,"></el-table-column-stub>
        <el-table-column-stub type="default" label="Telp" prop="phone" width="150" sortable="custom" resizable="true" filtermultiple="true" sortorders="ascending,descending,"></el-table-column-stub>
        <el-table-column-stub type="default" label="Role" prop="role_label" width="150" resizable="true" filtermultiple="true" sortorders="ascending,descending,"></el-table-column-stub>
        <el-table-column-stub type="default" label="Status" classname="status-col" prop="status" width="150px" sortable="custom" resizable="true" filtermultiple="true" sortorders="ascending,descending,"></el-table-column-stub>
        <el-table-column-stub type="default" label="Actions" width="250px" resizable="true" align="center" filtermultiple="true" sortorders="ascending,descending,"></el-table-column-stub>
      </el-table-stub>
      <pagination-stub total="0" page="1" limit="10" pagesizes="10,20,30,50" layout="total, sizes, prev, pager, next, jumper" background="true" autoscroll="true" style="display: none;"></pagination-stub>
    </el-col-stub>
  </el-row-stub>
</div>
`;
